src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -80,88 +77,121 @@ public sealed class SqlServerFhirModel : IStartable
+            ThrowIfNotInitialized();
 
             VLatest.QuantityCodeTable quantityCodeTable = VLatest.QuantityCode;
             return GetStringId(_quantityCodeToId, code, quantityCodeTable, quantityCodeTable.QuantityCodeId, quantityCodeTable.Value);
         }
 
         public bool TryGetQuantityCodeId(string code, out int quantityCodeId)
         {
-            ThrowIfNotStarted();
+            ThrowIfNotInitialized();
             return _quantityCodeToId.TryGetValue(code, out quantityCodeId);
         }
 
-        // TODO: Once this no longer implements IStartable, rename to "EnsureInitialized" (work item 75558).
-        public void Start()
+        public Task EnsureInitialized()
         {
-            _schemaInitializer.Start();
-
-            if (_searchParameterDefinitionManager is IStartable startable)
+            if (_schemaInformation.Current == null)
             {
-                startable.Start();
+                // TODO: What is the correct error to throw here?
+                throw new NullReferenceException("The current schema version should not be null");
             }
 
-            // As long as SqlServerFhirModel implements IStartable, avoid initialization if we are only running the base schema.
-            if (_schemaInformation.Current == null)
+            // If the fhir-server is just starting up, synchronize the fhir-server dictionaries with the SQL database
+            Initialize((int)_schemaInformation.Current, true);
+
+            return Task.CompletedTask;
+        }
+
+        public void Initialize(int version, bool runAllInitialization)
+        {
+            if (_highestInitializedVersion == version)
             {
                 return;
             }
 
             var connectionStringBuilder = new SqlConnectionStringBuilder(_configuration.ConnectionString);
+            _logger.LogInformation("Initializing {Server} {Database} to version {Version}", connectionStringBuilder.DataSource, connectionStringBuilder.InitialCatalog, version);
+
+            if (runAllInitialization)
+            {
+                InitializeV1();
 
-            _logger.LogInformation("Initializing {Server} {Database}", connectionStringBuilder.DataSource, connectionStringBuilder.InitialCatalog);
+                if (version >= 6)
+                {
+                    InitializeV6();
+                }
 
+                _highestInitializedVersion = version;
+            }
+            else
+            {
+                switch (version)
+                {
+                    case 1:
+                        InitializeV1();
+                        break;
+                    case 6:
+                        InitializeV6();
+                        break;
+                }
+
+                _highestInitializedVersion = version;
+            }
+        }
+
+        private void InitializeV1()
+        {
             using (var connection = new SqlConnection(_configuration.ConnectionString))
```

Comment:
Eventually this will need to use the connection factory for the managed idenity token, which will force this to become async :)

Reply:
Right thanks, I made a note to update this when I pull in master and the `IHostedService` changes into my feature branch 👍